In [ ]:
n = int(input('Please enter a number :'))
y = 0
for i in range(0,n+1):
    if i !=1:
        print(i, end=' ')
        print("Hello github")
        print("still not starting")


In [ ]:
#todo
user1=str(input('Please enter something : '))
bin = {'0','1'}
set_1 = set(user1)
if set_1==bin or bin==0 or bin==1:
  print('Binary Number')
else:
  print('Not a binary number')

Write a python program that takes 2 inputs from the user. It should print -
1. A series of numbers starting from the first input(X) to second input(Y) 
inclusive, with following terms which are the multiples of X.
2. All the numbers in every 3rd position of the original series .
3. Summation of all the numbers in every 3rd position of the original 
series.

NB: Both of the inputs, X and Y, can not be a negative number or 0. The 
second input(Y), must be greater than the first input(X). The program 
should print "Invalid input" otherwise.


Sample input 1:
4
28

Sample Output 1:
The original series: 4, 8, 12, 16, 20, 24, 28
Numbers in every 3rd position: 12, 24
Sum of all the numbers in every 3rd position: 36

Explanation 1:
The user entered 4 as the first input and 28 as the second input. So, the 
program prints the series of numbers that are multiples of first input(4) 
within the range of 4 and 28 inclusive: “The original series: 4, 8, 12, 16, 
20, 24, 28". Then the program prints the numbers in every 3rd position of 
that series. Here, the numbers 12 and 24 are in the 3rd and 6th position of 
the series. So, it should print: “Numbers in every 3rd position: 12,  24”. 
Finally, the program prints “Sum of all the numbers in every 3rd position: 
36” as 12 + 24 = 36


Sample input 2:
0
15

Sample Output 2:
Invalid input

Explanation 2:
The user enters 0 as the first input, which is not allowed. The program 
prints "Invalid input" and ends.

Sample input 3:
7
99

Sample Output 3:
The original series: 7, 14, 21, 28, 35, 42, 49, 56, 63, 70, 77, 84, 91, 98
Numbers in every 3rd position: 21, 42, 63, 84
Sum of all the numbers in every 3rd position: 210

Explanation 3:
The user entered 7 as the first input and 99 as the second input. So, the 
program prints the series of numbers that are multiple of first input(7) 
within the range of 7 and 99 inclusive: “The original series: 7, 14, 21, 
28, 35, 42, 49, 56, 63, 70, 77, 84, 91, 98". Then the program prints the 
numbers in every 3rd position of that series. So, it prints : “Numbers in 
every 3rd position: 21, 42, 63, 84”. Finally, the program prints “Sum of 
all the numbers in every 3rd position: 210”, as 21 + 42 + 63 + 84 = 210

In [ ]:
man=int(input('enter mango'))
lychee=int(input('enter lichu '))
org = int(input('enter komolabanu'))
sum = man*120+lychee*400+org*220
print('Total',sum)
if (sum<=450 and sum>=0):
  print(sum)
elif (sum>=451 and sum<=699):
  print(sum-25)
elif (sum>=700 and sum<=849):
  print(sum-50)
elif (sum>=850 and sum<=1599):
  print(sum-150)
elif (sum>=1600):
  print(sum-250)
else:
  print('error')

enter mango2
enter lichu 1
enter komolabanu3
Total 1300
1150


In [ ]:
x=int(input())
y=int(input())
org=0
for i in range(x,y+1,x):
  if(y==i):
    print(i,end='')
  else:
    print(i,end=' ,')
   

4
28
4 ,8 ,12 ,16 ,20 ,24 ,28

Write the python programs, which prints the following sequences of values in loops:

a) 24, 18, 12, 6, 0, -6
b) -10, -5, 0, 5, 10, 15, 20
c) 18, 27, 36, 45, 54, 63
d) 18, -27, 36, -45, 54, -63


In [ ]:
count=18
while count<=63:
  if count%2==0:
    print(count,end=',')
  elif count==63:
    print(count,end='')
  else:
    print(-1*count,end=',')
  count+=9

18,-27,36,-45,54,63

In [ ]:
count = 18
while count <=63:
  if count ==63:
    print(count)
    count+=9
  else:
    print(count,end=',')
    count+=9

18,27,36,45,54,63


In [ ]:
count = -10
while count <=20:
  if count==20:
    print(count)
    count+=5
  else:
    print(count,end=',')
    count+=5

-10,-5,0,5,10,15,20


In [ ]:
count = 24
while count>=-6:
  if count==-6:
    print(count,end='')
    count= count-6
  else:
    print(count,end=',')
    count-=6

24,18,12,6,0,-6

Task 2
Write a Python code for the following:

1) Ask the user to enter the name of his favorite car.
2) Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

==========================================================

Example 1: If the user enters “Toyota” and 2, your program should print the name “Toyota” two times.

Sample Input 1:
Toyota
2

Sample Output 1:
Toyota
Toyota

In [ ]:
car = input('Please enter car : ')
num = int(input('number de betta'))
count=0
while count<num:
  print(car)
  count+=1

Please enter car : kiki
number de betta2
kiki
kiki


Write the Python code of a program that adds all numbers that are multiples of both 7 and 9 up to 600 (including 600) i.e. 63, 126, 189, 252, ....

The output of your program should be: 2835
since 63 + 126 + 189 + 252 + 315 + 378 + 441 + 504 + 567 = 2835

In [ ]:
sum=0
for i in range(1,601):
  if i%7==0 and i%9==0:
    sum=sum+i
print(sum)


2835


Write the Python code of a program that adds all numbers that are multiples of either 7 or 9 up to 600 (including 600) i.e. 7, 9, 14, 18, 21, 27, 28, 35, 36.....
Ensure that numbers like 63, 126, 189 which are multiples of both 7 and 9 are added only once in the sum.

The output of your program should be: 42649

In [ ]:
sum = 0
for i in range(1,600):
  if i%7==0 or i%9==0:
    sum=sum+i
print(sum)

42649


Write a Python code of a program that adds all numbers that are multiples of either 7 or 9 but not both, up to 600 (including 600) i.e. 7, 9, 14, 18, 21..... and so on but not the numbers 63, 126, 189..... which are multiples of both 7 and 9.

The output of your program should be: 39814

In [ ]:
sum=0
for i in range(0,601):
  if i%7==0 and i%9!=0:
    sum=sum+i
  elif i%7!=0 and i%9==0:
    sum=sum+i
  else:
    pass
print(sum)

39814


Write the Python code of a program that displays all the odd numbers between 10 and 50 (inclusive).

Output: 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49

In [ ]:
for i in range(10,50):
  if i%2!=0:
    print(i,end=' ')


11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $

==========================================================

**Sample Input 1:**\
5

**Sample Output 1:**\
15

**Explanation:** y = 1 - 4 + 9 - 16 + 25 = 15

==========================================================

**Sample Input 2:**\
10

**Sample Output 2:**\
-55

**Explanation:** y = 1 - 4 + 9 - 16 + 25 - 36 + 49 - 64 + 81 - 100 = -55

==========================================================

**Sample Input 3:**\
20

**Sample Output 3:**\
-210



In [ ]:
n = int(input())
y=0
for i in range(0,n+1):
  if i%2==0:
    y=y-(i*i)
  else:
    y=y+(i*i)
print(y)
    

5
15


Write a Python code of a program that asks the user to enter ten numbers and then display the total and the average of ONLY the odd numbers among those ten numbers.
[Please do not use list for this task]

==========================================================

Sample Input 1:
1

2

3

4

5

6

7

8

9

10

Sample Output 1: The total of the odd numbers is 25 and their average is 5.0

In [ ]:
y=0
odd=0
for i in range(0,10):
  x=int(input())
  if x%2!=0:
    y=y+x
    odd+=1
print(y)
print(y/odd)

1
2
3
4
5
6
7
8
9
10
25
5.0


Task 9
Write a Python code for the following:

Ask the user to enter a Number, N
Display the summation of multiples of 7 up to that number (from 1 to N inclusive)
==========================================================

Sample Input 1:
50

Sample Output 1:
196

In [ ]:
sum=0
n=int(input())
for i in range(1,n+1):
  if i%7==0:
    sum=sum+i
print(sum)


50
196


Write a Python code that will read 5 numbers from the user. Your program should print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 5 numbers.

==========================================================

Sample Input 1:
1
2
3
4
5

Sample Output 1:
1
3
6
10
15

In [ ]:
sum=0
for i in range(1,6):
  n=int(input())
  sum=sum+i
  print(sum,end=' ')


1
1 2
3 3
6 4
10 5
15 

Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. (Right to Left)

[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]

Example: If the user gives 32768, then print 8, 6, 7, 2, 3

In [ ]:
n = int(input())
while n>0:
  if n<10: 
    print(n%10,end='') #10 diye mod korle baki thake 8 oi 8 print hoy
    n=n//10 # floor division korle 8 bad chole jay
  else:
    print(n%10,end=',')
    n=n//10

321
1,2,3

Write a Python program that takes a number and prints how many digits are in that number.
[Consider the input number to be an INTEGER.]

[You are not allowed to use len() function]

Example: If the user gives 9876, your program should print 4.

In [ ]:
n = int(input())
count=0
new=n
while n>0:
  count+=1
  n=n//10
#print(count)
while new>0:
  print(new//(10**(count-1)))
  y=new%(10**count-1)
  count-=1


123
1
12
123
1229.0


ZeroDivisionError: ignored

Write a Python program that takes a number from the user and prints its digits from left to right.

[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]

=========================================================================

Example: if the user gives 32768, then print 3, 2, 7, 6, 8

In [ ]:
n=int(input())
count=0
new=n
while n>0:
  count+=1
  n=n//10
while new>0:
  if new<10:
    print(new//(10**(count-1)))
    new=new%(10**(count-1))
    count-=1
  else:
    print(new//(10**(count-1)),end=',')
    new=new%(10**(count-1))
    count-=1
    


321
3,2,1


Write a Python program that takes a number as input from the user and prints the divisors of that number as well as how many divisors the number has.

==========================================================

Sample Input 1:
6

Sample Output 1:
1, 2, 3, 6
Total 4 divisors.

In [ ]:
n=int(input())
count=0
for i in range(1,n+1):
  if n%i==0 and n==i:
    print(i)
    count+=1
  elif n%i==0:
    print(i,end=',')
    count+=1
  else:
    pass
print('Total ',count,' divisors')



6
1,2,3,6
Total  4  divisors


Write a Python program that takes a number as input from the user and tells if it is a perfect number or not.

Perfect Number: An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

==========================================================

Sample Input 1:
6

Sample Output 2:
6 is a perfect number

In [ ]:
n=int(input())
sum=0
for i in range(1,n):
  if n%i==0:
    sum=sum+i
if sum==n:
  print('perfect')
else:
  print('not perfect')


33
not perfect


Write a Python program that asks the user for one number and tells if it is a prime number or not.

Prime Number: If a number has only two divisors, (1 and itself), then it is a prime number. If it is divisible by more numbers, then it is not a prime.

Hint: You may use the code for counting divisors from Task 14.

==========================================================

Sample Input 1:
11

Sample Output 1:
11 is a prime number

In [ ]:
n= int(input())
count=0
for i in range(1,n+1):
  if n%i==0:
    count+=1
if count==2:
  print('Prime number')
else:
  print('NOt a prime number')

6
NOt a prime number


Write a Python program that asks the user for a quantity, then takes that many numbers as input and prints the maximum, minimum and average of those numbers.

[Please note that you CANNOT use max, min built-in functions]

[Also, you DO NOT need to use lists for this task]

==========================================================

Example: If the user enters 5 as an input for quantity and the enters the 5 numbers, 10, 4, -1, -100, and 1.
The output of your program should be: “Maximum 10”, “Minimum -100”, “Average is -17.2” as shown below.

Input:
5 10 4 -1 -100 1

Output:
Maximum 10
Minimum -100
Average is -17.2

In [ ]:
q= int(input())
sum=0
max=-999999
min=999999
for i in range(q):
  n=int(input())
  sum=sum+n
  if n>max:
    max=n
  if n<min:
    min=n
print(max)
print(min)
print(sum/q)

1
2
2
2
2.0


Task 18
Write a python program that prints a square of size N using + where N will be given as input as illustrated in the examples below.

Hint: You may need to use nested loops to solve this problem.

==========================================================

Sample Input 1:

5

Sample Output 1:
+++++
+++++
+++++
+++++
+++++

In [ ]:
n=int(input())
for i in range(n):
  for j in range(n):
    print('+',end='')
  print()
  

5
+++++
+++++
+++++
+++++
+++++


Write a Python program that takes a number from the user and generates the
following pattern. If the user oven number Is NO greater than 4, then
print "Invalid input"
Sample Input1: 6
Sample Outputt:
######
88##8&
&8##＄8
＆&##8&
88#48&
Sample Input2: 3
Sample Output2:
Invalid input

In [ ]:
n=int(input())
if n>4:
  for row in range(1):
    for column in range(n):
      print('#',end='')
    print()
  for row in range(2,n):
    for column in range(0,2):
      print('&',end='')
    for column in range(2,4):
      print('#',end='')
    for column in range(4,6):
      print('&',end='')
    print()
  for row in range(n-1,n):
    for column in range(n):
      print('#',end='')
  print()
else:
  print('Invalid input')



6
######
&&##&&
&&##&&
&&##&&
&&##&&
######


In [ ]:
num = int(input("Enter a number:"))

if num>4:
    for i in range(num):
        for j in range(num):
            if i == 0 or i == (num - 1):
                print("#", end="")
            else:
                half = num // 2
                if num%2 != 0:

                    if j == (half-1) or j== (half +1) or j == (half):
                        print("#", end = "")
                    else:
                        print("&", end = "")
                else:

                    if j == (half) or j == (half-1):
                        print("#", end = "")
                    else:
                        print("&", end = "")
        print()
else:
    print("Invalid Number")

Enter a number:7
#######
&&###&&
&&###&&
&&###&&
&&###&&
&&###&&
#######


Write a python program that prints a rectangle of size M (height/line numbers) and N (length/column numbers) using incrementing numbers where M and N will be given as input. Please look at the examples below for clarification.

Hint: You may need to use nested loops and print the loop counter variable in one of the loops.

==========================================================

Sample Input 1:
4
6

Sample Output 1:
123456
123456
123456
123456

In [ ]:
row=int(input())
column=int(input())
for i in range(1,row):
  for j in range(1,column+1):
    print(j,end='')
  print()

4
6
123456
123456
123456


Sample Input 1:
4

Sample Output 1:
1
12
123
1234

In [ ]:
n=int(input())
for i range(1,n):
  print(i)
  

In [ ]:
n=int(input())
p=0
q=1
if n==1:
  print(p,end=' ')
elif n==2:
  print(p,end=' ')
  print(q,end=' ')
else:
  print(p,end=' ')
  print(q,end=' ')
  for i in range(1,n-1):
    r=p+q
    p=q
    q=r
    if r<n:
      print(r,end=' ')

10
0 1 1 2 3 5 8 

In [ ]:
dec= int(input())
for i in range(dec//2): #karon 2diye vag korle jotobar jaay totobar e to neya lagtese
  rem=dec%2
  dec=dec//2
  print(rem,end='')

10
01010

Task 23
Write a python program that converts a binary number to a decimal integer number after taking the binary number as an input from the user.

Note: A binary number can be converted to its corresponding decimal number by multiplying each binary digit with a power of 2, where the power denotes the position of the binary digit

==========================================================

Example 1:

For example, to convert 1010 to decimal, we can do this

23  x 1 +  22  x 0 +  21  x 1 +  20  x 0 = 10

Sample Input 1:
1010

Sample Output 1:
10

In [ ]:
bin = input('enter a number: ')
dec = 0
for i in bin:
    dec = dec*2 + int(i)
print (dec)


enter a number: 10101100
172


Task 1
Write a Python program that takes a String as an input from the user and prints that String in reverse order.

You cannot use the built-in reverse() function for this task.

=====================================================================

Sample Input 1:
CSE110

Sample Output 1:
011ESC

In [ ]:
n= input()
print(n[-1::-1])

CSE110
011ESC


Write a python program that takes 2 inputs from the user, where the first input is a string with length greater than 1. The second input is the index of the first given string from where you have to start reversing. After reversing the first input string from that index, print the new string back to the user. See samples below for clarification.

=====================================================================

Sample Input 1:

aBcd1234defg

5

Sample Output 3:

21dcBa34defg

In [ ]:
n=input()
x=int(input())
print(n[x::-1],end='')
print(n[x+1::1])


aBcd1234defg
5
21dcBa34defg


Write a program that takes a string as input and prints “Binary Number” if the string contains only 0s or 1s. Otherwise, print “Not a Binary Number”.

=====================================================================

Sample Input 1:
01101101101

Sample Output 1:
Binary Number

In [ ]:
n = input()
count=0
bin='01'
for i in n:
  if i not in bin:
    count=1
    break
  else:
    pass
if count==1:
  print('Not a binary number ')
else:
  print('Binary number')

10202
Not a binary number 


In [ ]:
n=input()
if 'er'in n:
  n=n.replace('er','est')
  print(n)
elif 'est' in n:
  print(n)
elif len(n)<3:
  print(n)
else:
  print(n,end='er')

stronger
strongest


In [ ]:
n=input()
for i in range(len(n)):
  for j in range(i+1):
    print(n[j],end='')
  print()

ban
b
ba
ban


In [ ]:
for i in range(1):
  for j in range(2):
    print(i)
    print(j)

0
0
0
1


In [ ]:
n = input()
for i in n:
  print(i,': ',ord(i))

hello
h :  104
e :  101
l :  108
l :  108
o :  111


In [ ]:
n=input()
for i in range(len(n)):
  x=ord(n[i])
  if x==122:
    print('a',end='')
  else:
    x=x+1
    print(chr(x),end='')


xyz
yza

In [ ]:
n=input()
for i in range(1,len(n)+1,2):
  x=ord(n[i])-32
  print(chr(x),end='')

abcd
BD

In [ ]:
n=input()
x=input()
for i in range(len(n)):
  if n[i]==x:
    print('')
  else:
    print(n[i],end='')

abc,nj
,
abc
nj

In [ ]:
n=input()
x=n[0]
y=n[0]
for i in n[1:]:
  if i!=x:
    y+=i
    x=i
print(y)

aaaaaaaaaaaaaaaaaaa
a


In [ ]:
x=input()
y=input()
if y in x:
  z=x.replace(y,'')
  print(z)
else:
  if len(x)>3:
    print(x[1:-1])
  else:
    print(x)

eren yeager
k
ren yeage


In [ ]:
s=input()
l=s.split(', ')


45
45


In [ ]:
n=input()
x='@'
for x in n:
  

In [ ]:
n=int(input())
fivehun=n//500
n=n%500
hun=n//100
n=n%100
fifty=n//50
n=n%50
twen=n//20
n=n%20
ten=n//10
n=n%10
five=n//5
n=n%5
two=n//2
n=n%2
one=n//1
print('500tk:',fivehun,'notes')
print('100 :',hun)
print('50 :',fifty)
print('20 : ',twen)
print('10 : ',ten)
print('2 : ',two)
print('one :',one)




1455
500tk: 2 notes
100 : 4
50 : 1
20 :  0
10 :  0
2 :  0
one : 0


In [ ]:
for i in range(24,-7,-6):
  if i ==-6:
    print(i)
  else:
    print(i,end=',')

24,18,12,6,0,-6


In [ ]:
x=input()
y=int(input())
for i in range(y):
  print(x)

toyota
5
toyota
toyota
toyota
toyota
toyota


In [ ]:
n=int(input())
while n>0:
  if n<10:
    print(n%10)
    n=n//10
  else:
    print(n%10,end=',')
    n=n//10

321
1,2,3


In [ ]:
n=int(input())
count=0
while n>0:
  n=n//10
  count+=1
print(count)

2020
4


In [ ]:
quantity = int(input('Please enter your number : '))
sum = 0
min=9999999
max=-999999
for i in range(quantity):
  num=int(input('Please enter number :'))
  sum = sum+num
  if num>max:
    max=num
  if num<min:
    min=num
avg=sum/quantity
print('Maximum ',max)
print('Minimum ',min)
print('Average ',avg)

Please enter your number : 5
Please enter number :1
Please enter number :2
Please enter number :3
Please enter number :4
Please enter number :5
Maximum  5
Minimum  1
Average  3.0


In [ ]:
n=int(input())
for i in range(n):
  for j in range(n):
    print('+',end='')
  print()

5
+++++
+++++
+++++
+++++
+++++


In [ ]:
x=int(input())
y=int(input())
for i in range(1,x+1):
  for j in range(1,y+1):
    print(j,end='')
  print()

4
5
12345
12345
12345
12345


In [ ]:
x=int(input())
for i in range(1,x+1):
  for j in range(1,i+1):
    print(j,end='')
  print()

5
1
12
123
1234
12345


In [ ]:
n=int(input())
p=0
q=1
if n==1:
  print(p,end=',')
elif n==2:
  print(p,end=',')
  print(q,end=',')
else:
    print(p,end=',')
    print(q,end=',')
    for i in range(3,n):
      r=p+q
      p=q
      q=r
      if r<n:
        print(r,end=',')
  

10
0,1,1,2,3,5,8,

In [ ]:
n=input()
x=int(input())
print(n[x::-1],end='')
print(n[x+1::1])

12345
2
32145


In [ ]:
n=input()
bin ='01'
count=0
for i in n:
  if i not in bin:
    count+=1
if count==0:
  print('binary')
else:
  print('not')

101010101013
not


In [ ]:
n=input()
if 'er' in n:
  x=n.replace('er','est')
  print(x)
elif 'est' in n:
  print(n)
elif len(n)>3:
  print(n+'er')
else:
  print(n)


est
est


In [ ]:
n=input()
for i in range(1,len(n)+1):
  for j in range(i):
    print(n[j],end='')
  print()

bangla
b
ba
ban
bang
bangl
bangla


In [ ]:
n=input()
for i in n:
  print(i,':',ord(i))

tarunno
t : 116
a : 97
r : 114
u : 117
n : 110
n : 110
o : 111


In [ ]:
n=input()
x=input()
if x in n:
  y=n.replace('a','')
  print(y)
elif len(n)<3:
  print(n)
else:
  print(n[1:-1:1])

hi
k
hi


In [ ]:
for i in range(1,6):
  for j in range(1,i+1):
    print(j,end=' ')
  print()

1 
1 2 
1 2 3 
1 2 3 4 
1 2 3 4 5 


In [ ]:
n=int(input())
sum=0
for i in range(1,n+1):
  sum+=i
print(sum)

10
55


In [ ]:
a=input().split(',')
print(a)
print(int(a[0])+int(a[1]))

In [ ]:
si="7,"
newl1=sm.split(',')
newl1=[]\
for i in newl1:
  newl1.append(int(i))
  print('my list ',newl1)

In [ ]:
#9 number 
x=[]
n=input()\
dig=''
for i in n:
  if j!=-' ':
    dig+=i
  else:
    x.append(int(dig))
    dig=''
x.appen(int(dig))
print(x)
for j in x:
  if i%2-==0:
    x.remove(i)
print(x)